code/processes/gateway.q

```diff
@@ -284,8 +284,13 @@ checktimeout:{
   finishquery[qids;1b;0Ni]];
  }
 
-
-
+// check if hdb query has date
+checkdate:{[query;servertype]
+ if[servertype=`hdb;
+  $[type [query] = 0h;
```

Comment:
I don't think the `$` should be used like this - I think what you should be doing here is using the `$` to get the query into a useful format so that regardless of the incoming type, you can handle it the same way. My suggestion would be something along the lines of:

```
query:$[10h=type query;parse;]query
```

which will parse the query if it's a string, or not if it's a list and then either way, it's a list going forward.

Also, unnecessary whitespace between `type` and `[query]`, plus by rearranging as in my suggestion you can get rid of square brackets